api/objects.proto

```diff
@@ -249,6 +249,14 @@ message Cluster {
 	// If the key is empty, the node will be unlocked (will not require a key
 	// to start up from a shut down state).
 	repeated EncryptionKey unlock_keys = 9;
+	
+	// RoutingMeshOff denotes whether the routing mesh is currently disabled.
+	// By default the routing mesh is enabled. As result of user removing the
+	// ingress network, this flag will be set to true. Cluster needs to carry
+	// this information, so that on reboot or leader re-election, new leader
+	// will not automatically create the routing mesh if it cannot find one
+	// in store. 
+	bool routingMeshOff = 10;
```

Comment:
I was suggesting moving ingress network creation out of the allocator, to where the initial cluster and node objects are created (see link). It kind of makes sense to do that, since there would be a single place where initial resources are created.

Every time the manager enters the leader state, it tries to create a default cluster object, because this might be a fresh swarm that has never had a leader before. We could also create an ingress network at this time, if creating the default cluster object succeeded (which means this is indeed a fresh swarm).